framework/include/base/SubProblem.h

```diff
@@ -348,6 +373,7 @@ class SubProblem : public Problem
    */
   void checkMatProps(std::map<unsigned int, std::set<std::string> > & props,
                      std::map<unsigned int, std::multimap<std::string, std::string> > & check_props,
+                     std::map<unsigned int, std::set<MaterialPropertyName> > & zero_props,
```

Comment:
OK - Agreed, let's take care of that later.

On Mon, Mar 14, 2016 at 9:21 AM Daniel Schwen <notifications@github.com>
wrote:

> In framework/include/base/SubProblem.h
> <https://github.com/idaholab/moose/pull/6567#discussion_r56017549>:
>
> > @@ -348,6 +373,7 @@ class SubProblem : public Problem
> >     */
> >    void checkMatProps(std::map<unsigned int, std::set<std::string> > & props,
> >                       std::map<unsigned int, std::multimap<std::string, std::string> > & check_props,
> > +                     std::map<unsigned int, std::set<MaterialPropertyName> > & zero_props,
>
> Well, this is a bit more involved than just changing types. Both, block
> and boundary, variants are passed into checkMatProps. To switch over to
> the correct types this function would need to get templated! We then could
> remove the type string parameter entirely and replace it with a template
> specialized helper function. Would be a bit more elegant but I thought it
> was beyond the scope of this PR.
> I've noticed the convolution of issues blowing up certain other PRs, which
> makes them rather cumbersome to review. I'd like to avoid this. I'd be
> happy to address this in another PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6567/files#r56017549>.
>